identity_storage/src/key_id_storage/method_digest.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_verification::VerificationMethod;
+use seahash::SeaHasher;
+use std::hash::Hasher;
+
+/// Unique description through hashing of key materials in `VerificationMethod`s.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct MethodDigest {
+  /// Version of hashing.
+  pub version: u8,
+  /// Hash value.
+  pub value: u64,
+}
+
+impl MethodDigest {
+  /// Creates a new [`MethodDigest`].
+  pub fn new(verification_method: &VerificationMethod) -> identity_verification::Result<Self> {
+    let mut hasher: SeaHasher = SeaHasher::new();
+    let fragment = verification_method
+      .id()
+      .fragment()
+      .ok_or(identity_verification::Error::MissingIdFragment)?;
+
+    let method_data: Vec<u8> = verification_method.data().try_decode()?;
+    hasher.write(fragment.as_bytes());
+    hasher.write(&method_data);
+    let key_hash: u64 = hasher.finish();
+    Ok(Self {
+      version: 0,
+      value: key_hash,
+    })
+  }
+}
```

Comment:
Let's add a test case here in the style of this one: https://github.com/iotaledger/identity.rs/blob/support/v0.6/identity_account_storage/src/types/key_location.rs#L106

Except you can just create a hardcoded verification method, create its digest resulting in value `X`, then add a test ensuring that the hardcoded verification method always hashes to `X`. That same method should result in the same `MethodDigest` in a Wasm test. I think we got rid of `wasm.rs` in `bindings/wasm/tests`, but it would be great if we could test that.